plugins/auth/irods_auth_plugin-native/libirods_auth_plugin-native.cpp

```diff
@@ -0,0 +1,467 @@
+#include "authentication_plugin_framework.hpp"
+
+#include "base64.h"
+
+int get64RandomBytes( char *buf );
+void setSessionSignatureClientside( char* _sig );
+void _rsSetAuthRequestGetChallenge( const char* _c );
+
+namespace
+{
+    using json = nlohmann::json;
+
+    auto request(rcComm_t* comm, const json& msg)
+    {
+        auto str = msg.dump();
+
+        bytesBuf_t inp;
+        inp.buf = (void*)str.c_str();
```

Comment:
consider using a c++ cast instead of a c-style cast.

Reply:
Will do